pkg/reconciler/route/resources/service.go

```diff
@@ -112,7 +112,11 @@ func makeK8sService(ctx context.Context, route *v1.Route, targetName string) (*c
 				// This service is owned by the Route.
 				*kmeta.NewControllerRef(route),
 			},
-			Labels:      kmeta.UnionMaps(route.GetLabels(), svcLabels),
+			Labels: kmeta.FilterMap(kmeta.UnionMaps(route.GetLabels(), svcLabels), func(key string) bool {
```

Comment:
Do we know where they come from? Since filter then union is probably more effective.

Reply:
Yes, we known it is from Route. Thank you. Updated.